### DISCUSSION TITLE
Feature request: Add support for arbitrary-palette positional dithering

### DISCUSSION THREAD

**Discussion by Fernap** [ImageMagick/ImageMagick#6696] [Contribute]
### Prerequisites

- [X] I have written a descriptive issue title
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported
- [x] I have verified that I am using the latest version of ImageMagick

### ImageMagick version

7

### Operating system

Windows

### Operating system, version and so on

-

### Description

I'm new here, so I hope this is a reasonable way to request a feature.  I think it would be nice to add the support for positional (ordered) dithering with an arbitrary palette (either picked by ImageMagick, or user-supplied).

Some rather old IM doc about color quantization [(here, all the way at the very bottom)](https://legacy.imagemagick.org/Usage/quantize/) even talks about this very possibility, but laments the lack of a suitable algorithm to use.  However, Joel Yliluoma has given a fairly detailed description of just such an algorithm in a [writeup here](https://bisqwit.iki.fi/story/howto/dither/jy/) that seems like it would be appropriate.

This should help to produce nicer-looking animations than error-diffusing methods, and would be suitable for small palettes.  Unfortunately, this is all a bit beyond my skill level, so I'm hoping someone else might be able to implement it.

### Steps to Reproduce

-

**Comment by urban-warrior** [comment]
The ImageMagick development will likely not implement the arbitrary-palette positional dithering algorithm.  However, we would accept a pull request from a user that wants to contribute the algorithm.
